### DISCUSSION TITLE
hexo-theme-fluid 无法在文章页使用 waline

### DISCUSSION THREAD

**Discussion by amakerlife** [walinejs/waline#1651] [Q&A]
#### 请确认
- [x] 是当前最新的 [Release 版本](https://github.com/fluid-dev/hexo-theme-fluid/releases)
- [x] 本地 `hexo clean && hexo s`，并且清除浏览器缓存，仍可复现
- [ ] 已经排除是其他 Hexo 插件影响

#### Bug 描述

在文章页中，waline 无法正常工作，配置如下图。但奇怪的是，友链页却可以。

![](https://img.makerlife.tk/data/20221211220433.png)

waline 静态：<https://cdn.bootcdn.net/ajax/libs/waline/2.14.1/>

控制台无报错

浏览器：Chrome

#### 复现步骤
可在该文章中复现：<https://www.makerlife.tk/post/solution-AT1898/>

#### 截图

见上

**Comment by gumpBOBO** [comment]
comments:
    enable: true
    # 指定的插件，需要同时设置对应插件的必要参数
    # The specified plugin needs to set the necessary parameters at the same time
    # Options: utterances | disqus | gitalk | valine | waline | changyan | livere | remark42 | twikoo | cusdis
    type: waline

post的配置中 评论这里要打开

**Reply by amakerlife** [reply]
已经打开，无效，见 hexo-theme-fluid issue#893

**Comment by undefined-hestudio** [comment]
![image](https://user-images.githubusercontent.com/119711513/211178583-413df379-b377-43c4-82b3-53f5385725c9.png)

不是能用吗？

**Reply by amakerlife** [reply]
![](https://img.makerlife.tk/data/202301081745981.png)

你再刷新试试

有时候可以大部分时候一直在转圈

**Reply by undefined-hestudio** [reply]
如果加载出评论系统的基本界面，则说明你的配置文件完全正确。

如果你是在vercel部署的服务端，因为vercel在中国大陆的链接速度就像裸连GitHub一样，大多数时间是无法访问的，这就使得在中国大陆访问的用户在多数时间访问不了。

建议使用国内的一些云函数平台或者私有部署，毕竟你的用户不会挂着梯子看你的博客。你需要保证你的服务端在中国大陆能做到正常访问。

如果你的博客是在服务器部署，那么我个人建议你将服务端和数据库完全部署到你的服务器上。

**Reply by amakerlife** [reply]
> 如果加载出评论系统的基本界面，则说明你的配置文件完全正确。
> 
> 如果你是在vercel部署的服务端，因为vercel在中国大陆的链接速度就像裸连GitHub一样，大多数时间是无法访问的，这就使得在中国大陆访问的用户在多数时间访问不了。
> 
> 建议使用国内的一些云函数平台或者私有部署，毕竟你的用户不会挂着梯子看你的博客。你需要保证你的服务端在中国大陆能做到正常访问。
> 
> 如果你的博客是在服务器部署，那么我个人建议你将服务端和数据库完全部署到你的服务器上。

你可以看一下我的服务端地址，使用的是自定义域名+cf cdn，由于 gfw 对 vercel 的封锁方式是 dns 污染，所以我的域名在中国大陆是完全可以访问的，且在无梯子的情况下就是如此。

配置文件和服务端都是完全正确的，且在友链页加载出评论的机率基本为 100%，文章页却做不到

**Reply by undefined-hestudio** [reply]
友链页能访问那就表明客户端没有问题，建议你先更新一下服务端程序
![Screenshot_2023-01-09-00-08-23-58_df198e732186825c8df26e3c5a10d7cd.jpg](https://user-images.githubusercontent.com/119711513/211206978-59e86096-7876-4e56-94e7-02b5c3fba6cd.jpg)

**Reply by amakerlife** [reply]
> 友链页能访问那就表明客户端没有问题，建议你先更新一下服务端程序 ![Screenshot_2023-01-09-00-08-23-58_df198e732186825c8df26e3c5a10d7cd.jpg](https://user-images.githubusercontent.com/119711513/211206978-59e86096-7876-4e56-94e7-02b5c3fba6cd.jpg)

服务端以更新至最新版本，仍无效

**Reply by undefined-hestudio** [reply]
我试着裸连了一下你的服务端，访问速度真的很慢，应该是`timeout`了，建议在国内加速一下，`waline.min.js`下载后放到网站里面，数据库也迁到国内。

即使你使用绑定域名实现绕过dns污染，但是vercel的服务器的地理位置本身就导致延迟过高。

**Reply by amakerlife** [reply]
> 我试着裸连了一下你的服务端，访问速度真的很慢，应该是`timeout`了，建议在国内加速一下，`waline.min.js`下载后放到网站里面，数据库也迁到国内。
> 
> 即使你使用绑定域名实现绕过dns污染，但是vercel的服务器的地理位置本身就导致延迟过高。

实测直连应该不会超时，waline.min.js 就是使用的国内源啊😂
